modal/functions.py

```diff
@@ -772,11 +773,18 @@ async def _load(self, client, stub, app_id, loader, message_callback, existing_f
             # Use cloudpickle. Used when working w/ Jupyter notebooks.
             # serialize at _load time, not function decoration time
             # otherwise we can't capture a surrounding class for lifetime methods etc.
-            function_serialized = cloudpickle.dumps(self._raw_f)
-            logger.debug(f"Serializing {self._raw_f.__qualname__}, size is {len(function_serialized)}")
+            function_serialized = self._info.serialized_function
             mod = inspect.getmodule(self._raw_f)
 
-            cls, _ = load_function_from_module(mod, self._raw_f.__qualname__)
+            try:
+                cls, _ = load_function_from_module(mod, self._raw_f.__qualname__)
+            except LocalFunctionError:
+                # if a serialized function is defined within a function scope
+                # we can't load it from the module and detect its parent class
+                # TODO: fix this somehow... maybe put the decorator on the
+                #       class instead for entrypoint classes
+                cls = None
```

Comment:
To some degree yeah, it's more of a graceful fallback for an error we can't currently handle for serialized functions. Would you prefer if I send in a flag whether to raise or not in case the class can't be loaded? Or some other structure suggestions?

Reply:
Don't think it's super important, but code like this looks like we're "suppressing" an error which I think can make debugging harder in some cases... that's why I don't love using exceptions for control flow. But it's a very minor thing.